### ISSUE TITLE
[core] Actor call hanging when actor died.

### ISSUE THREAD

**Issue by iycheng** [ray-project/ray#26772]
### What happened + What you expected to happen

https://gist.github.com/simon-mo/1d53713e65be945af722daa5807af6f8#file-app-py

This script is not supposed to hang given that it'll have multiple replicas. The application should be able to receive an error, but not hangs.

### Versions / Dependencies

master

### Reproduction script

https://gist.github.com/simon-mo/1d53713e65be945af722daa5807af6f8#file-app-py

### Issue Severity

High: It blocks me from completing my task.

**Comment by iycheng** [comment]
There are two issues observed:

1. The actor died notification is pushed late. The cause is about the GCS detect it late. Still checking about why.
2. Actor submitting hangs which shouldn't. It'll be unblocked until 1) happens.

**Comment by iycheng** [comment]
```
[2022-07-20 22:47:20,617 D 16281 16363] core_worker.cc:1920: Submitting actor task Type=ACTOR_TASK, Language=PYTHON, Resources: {}, function_descriptor={type=PythonFunctionDescriptor, module_name=ray.serve.replica, class_name=ServeReplica:DAGDriver, function_name=handle_request, function_hash=}, task_id=0a7603de59588c07fc54eb2aff11565b5138d2ad01000000, task_name=ServeReplica:DAGDriver.handle_request(), job_id=01000000, num_args=4, num_returns=3, depth=0, actor_task_spec={actor_id=fc54eb2aff11565b5138d2ad01000000, actor_caller_id=ffffffffffffffff2ae17fb78f14dd9fda33b67301000000, actor_counter=208}
[2022-07-20 22:47:20,617 D 16281 16363] task_manager.cc:36: Adding pending task 0a7603de59588c07fc54eb2aff11565b5138d2ad01000000 with 0 retries
[2022-07-20 22:47:20,617 D 16281 16363] reference_count.cc:200: Adding owned object 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:47:20,617 D 16281 16363] reference_count.cc:200: Adding owned object 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000
[2022-07-20 22:47:20,617 D 16281 16363] reference_count.cc:335: Increment ref count for submitted task argument fffffffffffffffffc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:47:20,617 D 16281 16363] direct_actor_task_submitter.cc:76: Submitting task 0a7603de59588c07fc54eb2aff11565b5138d2ad01000000
[2022-07-20 22:47:20,617 D 16281 16351] direct_actor_task_submitter.cc:423: Pushing task 0a7603de59588c07fc54eb2aff11565b5138d2ad01000000 to actor fc54eb2aff11565b5138d2ad01000000 actor counter 208 seq no 208 num queued 1
[2022-07-20 22:47:20,774 D 16281 16351] gcs_rpc_client.h:514: GCS channel status: 2
[2022-07-20 22:47:20,853 D 16281 16351] accessor.cc:784: Adding profile data, component type = worker, node id = 6fee9ec66df6acc50d14ace7c4963e6f98bd1b798515a5c120998220
[2022-07-20 22:47:20,854 D 16281 16351] profiling.cc:87: Pushed 1 events to GCS.
[2022-07-20 22:47:20,854 D 16281 16351] accessor.cc:795: Finished adding profile data, status = OK, component type = worker, node id = 6fee9ec66df6acc50d14ace7c4963e6f98bd1b798515a5c120998220
[2022-07-20 22:47:21,774 D 16281 16351] gcs_rpc_client.h:514: GCS channel status: 2
[2022-07-20 22:47:22,774 D 16281 16351] gcs_rpc_client.h:514: GCS channel status: 2
[2022-07-20 22:47:22,970 D 16281 16351] subscriber.cc:365: Long polling request has replied from 9ab6b0b9f58b9c57e7dd47768badd40d0f517b9ea00d344e000c4416
[2022-07-20 22:47:22,970 D 16281 16351] subscriber.cc:348: Make a long polling request to 9ab6b0b9f58b9c57e7dd47768badd40d0f517b9ea00d344e000c4416
[2022-07-20 22:47:22,970 I 16281 16351] accessor.cc:608: Received notification for node id = a61db2a59607330d55aa726dc21c65418ee4d88faae17379c6657666, IsAlive = 0
[2022-07-20 22:47:22,970 I 16281 16351] core_worker.cc:697: Node failure from a61db2a59607330d55aa726dc21c65418ee4d88faae17379c6657666. All objects pinned on that node will be lost if object reconstruction is not enabled.
[2022-07-20 22:47:22,970 D 16281 16351] reference_count.cc:1387: Published message for 1fa02b688fa4c7c15f40c38b4ea7145ed1c52be50100000001000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: 3, and primary node ID: NIL_ID, pending creation? 0
[2022-07-20 22:47:22,970 D 16281 16351] reference_count.cc:1387: Published message for fcc9f745b3554eb561393dc0e76fe0933af594230100000001000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: 3, and primary node ID: NIL_ID, pending creation? 0
[2022-07-20 22:47:22,970 D 16281 16351] reference_count.cc:1387: Published message for 9d887b46bfeb9a3c61393dc0e76fe0933af594230100000001000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: 3, and primary node ID: NIL_ID, pending creation? 0
[2022-07-20 22:47:22,970 D 16281 16351] reference_count.cc:1387: Published message for a8ab4186993562df5f40c38b4ea7145ed1c52be50100000001000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: 3, and primary node ID: NIL_ID, pending creation? 0
[2022-07-20 22:47:22,970 D 16281 16351] reference_count.cc:1387: Published message for b1998724544dff25fc54eb2aff11565b5138d2ad0100000001000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: 3, and primary node ID: NIL_ID, pending creation? 0
[2022-07-20 22:47:22,970 D 16281 16351] reference_count.cc:1387: Published message for c76493562716f40a61393dc0e76fe0933af594230100000001000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: 3, and primary node ID: NIL_ID, pending creation? 0
[2022-07-20 22:47:22,970 D 16281 16351] reference_count.cc:1387: Published message for b580a4e70e3fd60e5f40c38b4ea7145ed1c52be50100000001000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: 3, and primary node ID: NIL_ID, pending creation? 0
[2022-07-20 22:47:22,970 D 16281 16351] reference_count.cc:1387: Published message for 6eab9a1d67862b5061393dc0e76fe0933af594230100000001000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: 3, and primary node ID: NIL_ID, pending creation? 0
....
[2022-07-20 22:47:35,775 D 16281 16351] gcs_rpc_client.h:514: GCS channel status: 2
[2022-07-20 22:47:36,775 D 16281 16351] gcs_rpc_client.h:514: GCS channel status: 2
[2022-07-20 22:47:36,970 D 16281 16351] subscriber.cc:365: Long polling request has replied from 9ab6b0b9f58b9c57e7dd47768badd40d0f517b9ea00d344e000c4416
[2022-07-20 22:47:36,970 D 16281 16351] subscriber.cc:348: Make a long polling request to 9ab6b0b9f58b9c57e7dd47768badd40d0f517b9ea00d344e000c4416
[2022-07-20 22:47:36,970 I 16281 16351] accessor.cc:608: Received notification for node id = 62881a3a70d165da0ee5b0e02cf447d1029e8af831cb409ed3c8e279, IsAlive = 0
[2022-07-20 22:47:36,970 I 16281 16351] core_worker.cc:697: Node failure from 62881a3a70d165da0ee5b0e02cf447d1029e8af831cb409ed3c8e279. All objects pinned on that node will be lost if object reconstruction is not enabled.
[2022-07-20 22:47:36,970 D 16281 16351] reference_count.cc:1387: Published message for 1fa02b688fa4c7c15f40c38b4ea7145ed1c52be50100000001000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: 3, and primary node ID: NIL_ID, pending creation? 0
[2022-07-20 22:47:36,970 D 16281 16351] reference_count.cc:1387: Published message for fcc9f745b3554eb561393dc0e76fe0933af594230100000001000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: 3, and primary node ID: NIL_ID, pending creation? 0
.....
[2022-07-20 22:47:38,168 D 16281 16363] reference_count.cc:548: REF 5217a31b15284939fc54eb2aff11565b5138d2ad0100000001000000 borrowers: 0 local_ref_count: 0 submitted_count: 0 contained_in_owned: 0 contained_in_borrowed: 0 contains: 0 stored_in: 0 lineage_ref_count: 0
[2022-07-20 22:47:38,168 D 16281 16363] reference_count.cc:585: Deleting Reference to object 5217a31b15284939fc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:47:38,168 D 16281 16363] reference_count.cc:404: Releasing lineage for object 5217a31b15284939fc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:47:38,168 D 16281 16363] task_manager.cc:547: No lineage for object 5217a31b15284939fc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:47:38,168 D 16281 16363] reference_count.cc:316: Remove local reference 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:47:38,168 D 16281 16363] reference_count.cc:317: REF 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000001000000 borrowers: 0 local_ref_count: 0 submitted_count: 0 contained_in_owned: 0 contained_in_borrowed: 0 contains: 0 stored_in: 0 lineage_ref_count: 0
[2022-07-20 22:47:38,168 D 16281 16363] reference_count.cc:542: Attempting to delete object 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:47:38,168 D 16281 16363] reference_count.cc:548: REF 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000001000000 borrowers: 0 local_ref_count: 0 submitted_count: 0 contained_in_owned: 0 contained_in_borrowed: 0 contains: 0 stored_in: 0 lineage_ref_count: 0
[2022-07-20 22:47:38,168 D 16281 16363] reference_count.cc:585: Deleting Reference to object 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:47:38,168 D 16281 16363] reference_count.cc:404: Releasing lineage for object 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:47:38,168 D 16281 16363] task_manager.cc:551: Plasma object 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000001000000 out of scope
[2022-07-20 22:47:38,168 D 16281 16363] task_manager.cc:553: Task 0a7603de59588c07fc54eb2aff11565b5138d2ad01000000 has 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000001000000 in scope
[2022-07-20 22:47:38,168 D 16281 16363] task_manager.cc:553: Task 0a7603de59588c07fc54eb2aff11565b5138d2ad01000000 has 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000 in scope
[2022-07-20 22:47:38,168 D 16281 16363] task_manager.cc:556: Task 0a7603de59588c07fc54eb2aff11565b5138d2ad01000000 now has 1 plasma returns in scope
[2022-07-20 22:47:38,168 D 16281 16363] reference_count.cc:316: Remove local reference 7763988431ce6479fc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:47:38,168 D 16281 16363] reference_count.cc:317: REF 7763988431ce6479fc54eb2aff11565b5138d2ad0100000001000000 borrowers: 0 local_ref_count: 0 submitted_count: 0 contained_in_owned: 0 contained_in_borrowed: 0 contains: 0 stored_in: 0 lineage_ref_count: 0
[2022-07-20 22:47:38,168 D 16281 16363] reference_count.cc:542: Attempting to delete object 7763988431ce6479fc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:47:38,168 D 16281 16363] reference_count.cc:548: REF 7763988431ce6479fc54eb2aff11565b5138d2ad0100000001000000 borrowers: 0 local_ref_count: 0 submitted_count: 0 contained_in_owned: 0 contained_in_borrowed: 0 contains: 0 stored_in: 0 lineage_ref_count: 0
....
[2022-07-20 22:48:53,781 D 16281 16351] gcs_rpc_client.h:514: GCS channel status: 2
[2022-07-20 22:48:54,757 D 16281 16351] subscriber.cc:365: Long polling request has replied from 9ab6b0b9f58b9c57e7dd47768badd40d0f517b9ea00d344e000c4416
[2022-07-20 22:48:54,757 D 16281 16351] subscriber.cc:348: Make a long polling request to 9ab6b0b9f58b9c57e7dd47768badd40d0f517b9ea00d344e000c4416
[2022-07-20 22:48:54,757 I 16281 16351] actor_manager.cc:214: received notification on actor, state: DEAD, actor_id: fc54eb2aff11565b5138d2ad01000000, ip address: 172.31.58.40, port: 42767, worker_id: 405f212b30d9c3f62993d4d883b65d5b854216dbe1d4071da516075a, raylet_id:
c22791b03b5e4f31b7aca81879fd321d42790747fc8c383d000edc99, num_restarts: 0, death context type=ActorDiedErrorContext
[2022-07-20 22:48:54,757 D 16281 16351] actor_manager.cc:328: Actor name cache is invalidated for the actor of name SERVE_REPLICA::DAGDriver#XnytyC namespace serve id fc54eb2aff11565b5138d2ad01000000
[2022-07-20 22:48:54,757 D 16281 16351] direct_actor_task_submitter.cc:237: Disconnecting from actor fc54eb2aff11565b5138d2ad01000000, death context type=ActorDiedErrorContext
[2022-07-20 22:48:54,757 I 16281 16351] direct_actor_task_submitter.cc:270: Failing pending tasks for actor fc54eb2aff11565b5138d2ad01000000 because the actor is already dead.
[2022-07-20 22:48:54,757 I 16281 16351] direct_actor_task_submitter.cc:291: Failing tasks waiting for death info, size=0, actor_id=fc54eb2aff11565b5138d2ad01000000
[2022-07-20 22:48:54,757 D 16281 16351] task_manager.cc:459: Task attempt 288fc4e3db68adeafc54eb2aff11565b5138d2ad01000000 failed with error ACTOR_DIED
[2022-07-20 22:48:54,757 D 16281 16351] task_manager.cc:406: Task 288fc4e3db68adeafc54eb2aff11565b5138d2ad01000000 failed with error ACTOR_DIED
[2022-07-20 22:48:54,757 I 16281 16351] task_manager.cc:432: Task failed: IOError: Fail all inflight tasks due to actor state change.: Type=ACTOR_TASK, Language=PYTHON, Resources: {}, function_descriptor={type=PythonFunctionDescriptor, module_name=ray.serve.replica, cla
ss_name=ServeReplica:DAGDriver, function_name=handle_request, function_hash=}, task_id=288fc4e3db68adeafc54eb2aff11565b5138d2ad01000000, task_name=ServeReplica:DAGDriver.handle_request(), job_id=01000000, num_args=4, num_returns=3, depth=0, actor_task_spec={actor_id=fc5
4eb2aff11565b5138d2ad01000000, actor_caller_id=ffffffffffffffff2ae17fb78f14dd9fda33b67301000000, actor_counter=207}
[2022-07-20 22:48:54,757 D 16281 16351] reference_count.cc:1387: Published message for 288fc4e3db68adeafc54eb2aff11565b5138d2ad0100000002000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: -1, and primary node ID: NIL_ID, pending creation? 0
[2022-07-20 22:48:54,757 D 16281 16351] reference_count.cc:864: Merging ref fffffffffffffffffc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:48:54,757 D 16281 16351] reference_count.cc:444: Releasing ref for submitted task argument fffffffffffffffffc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:48:54,757 D 16281 16351] task_manager.cc:596: Treat task as failed. task_id: 288fc4e3db68adeafc54eb2aff11565b5138d2ad01000000, error_type: ACTOR_DIED
[2022-07-20 22:48:54,757 D 16281 16351] memory_store.cc:201: Putting object into memory store. objectid is 288fc4e3db68adeafc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:48:54,757 D 16281 16351] memory_store.cc:201: Putting object into memory store. objectid is 288fc4e3db68adeafc54eb2aff11565b5138d2ad0100000002000000
[2022-07-20 22:48:54,757 D 16281 16351] reference_count.cc:259: Add local reference 288fc4e3db68adeafc54eb2aff11565b5138d2ad0100000002000000
[2022-07-20 22:48:54,757 D 16281 16351] reference_count.cc:260: REF 288fc4e3db68adeafc54eb2aff11565b5138d2ad0100000002000000 borrowers: 0 local_ref_count: 2 submitted_count: 0 contained_in_owned: 0 contained_in_borrowed: 0 contains: 0 stored_in: 0 lineage_ref_count: 0
[2022-07-20 22:48:54,758 D 16281 16351] reference_count.cc:316: Remove local reference 288fc4e3db68adeafc54eb2aff11565b5138d2ad0100000002000000
[2022-07-20 22:48:54,758 D 16281 16351] reference_count.cc:317: REF 288fc4e3db68adeafc54eb2aff11565b5138d2ad0100000002000000 borrowers: 0 local_ref_count: 1 submitted_count: 0 contained_in_owned: 0 contained_in_borrowed: 0 contains: 0 stored_in: 0 lineage_ref_count: 0
[2022-07-20 22:48:54,758 D 16281 16351] reference_count.cc:321: REF 288fc4e3db68adeafc54eb2aff11565b5138d2ad0100000002000000 borrowers: 0 local_ref_count: 1 submitted_count: 0 contained_in_owned: 0 contained_in_borrowed: 0 contains: 0 stored_in: 0 lineage_ref_count: 0
[2022-07-20 22:48:54,758 D 16281 16351] memory_store.cc:201: Putting object into memory store. objectid is 288fc4e3db68adeafc54eb2aff11565b5138d2ad0100000003000000
[2022-07-20 22:48:54,758 D 16281 16351] task_manager.cc:459: Task attempt 0a7603de59588c07fc54eb2aff11565b5138d2ad01000000 failed with error ACTOR_DIED
[2022-07-20 22:48:54,758 D 16281 16351] task_manager.cc:406: Task 0a7603de59588c07fc54eb2aff11565b5138d2ad01000000 failed with error ACTOR_DIED
[2022-07-20 22:48:54,758 I 16281 16351] task_manager.cc:432: Task failed: IOError: Fail all inflight tasks due to actor state change.: Type=ACTOR_TASK, Language=PYTHON, Resources: {}, function_descriptor={type=PythonFunctionDescriptor, module_name=ray.serve.replica, class_name=ServeReplica:DAGDriver, function_name=handle_request, function_hash=}, task_id=0a7603de59588c07fc54eb2aff11565b5138d2ad01000000, task_name=ServeReplica:DAGDriver.handle_request(), job_id=01000000, num_args=4, num_returns=3, depth=0, actor_task_spec={actor_id=fc54eb2aff11565b5138d2ad01000000, actor_caller_id=ffffffffffffffff2ae17fb78f14dd9fda33b67301000000, actor_counter=208}
[2022-07-20 22:48:54,758 D 16281 16351] reference_count.cc:1387: Published message for 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: -1, and primary node ID: NIL_ID, pending creation? 0
[2022-07-20 22:48:54,758 D 16281 16351] reference_count.cc:864: Merging ref fffffffffffffffffc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:48:54,758 D 16281 16351] reference_count.cc:444: Releasing ref for submitted task argument fffffffffffffffffc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:48:54,758 D 16281 16351] task_manager.cc:596: Treat task as failed. task_id: 0a7603de59588c07fc54eb2aff11565b5138d2ad01000000, error_type: ACTOR_DIED
[2022-07-20 22:48:54,758 D 16281 16351] memory_store.cc:201: Putting object into memory store. objectid is 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000001000000
[2022-07-20 22:48:54,758 D 16281 16351] memory_store.cc:201: Putting object into memory store. objectid is 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000
[2022-07-20 22:48:54,758 D 16281 16351] reference_count.cc:259: Add local reference 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000
[2022-07-20 22:48:54,758 D 16281 16351] reference_count.cc:260: REF 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000 borrowers: 0 local_ref_count: 2 submitted_count: 0 contained_in_owned: 0 contained_in_borrowed: 0 contains: 0 stored_in: 0 lineage_ref_count: 0
[2022-07-20 22:48:54,758 D 16281 16351] reference_count.cc:316: Remove local reference 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000
[2022-07-20 22:48:54,758 D 16281 16351] reference_count.cc:317: REF 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000 borrowers: 0 local_ref_count: 1 submitted_count: 0 contained_in_owned: 0 contained_in_borrowed: 0 contains: 0 stored_in: 0 lineage_ref_count: 0
[2022-07-20 22:48:54,758 D 16281 16351] reference_count.cc:321: REF 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000 borrowers: 0 local_ref_count: 1 submitted_count: 0 contained_in_owned: 0 contained_in_borrowed: 0 contains: 0 stored_in: 0 lineage_ref_count: 0
[2022-07-20 22:48:54,758 D 16281 16351] memory_store.cc:201: Putting object into memory store. objectid is 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000003000000
[2022-07-20 22:48:54,758 D 16281 16351] task_manager.cc:226: Completing task ef82c691a2b0098585c59ffc623abb44d9968a5d01000000
[2022-07-20 22:48:54,758 D 16281 16351] reference_count.cc:1387: Published message for ef82c691a2b0098585c59ffc623abb44d9968a5d0100000001000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: 1469, and primary node ID: NIL_ID, pending creation? 1
[2022-07-20 22:48:54,758 D 16281 16351] task_manager.cc:249: Task return object ef82c691a2b0098585c59ffc623abb44d9968a5d0100000001000000 has size 1469
[2022-07-20 22:48:54,758 D 16281 16351] memory_store.cc:201: Putting object into memory store. objectid is ef82c691a2b0098585c59ffc623abb44d9968a5d0100000001000000
...
[2022-07-20 22:48:54,809 D 16281 16363] direct_actor_task_submitter.cc:76: Submitting task 4e1852778ed2b1e261393dc0e76fe0933af5942301000000
[2022-07-20 22:48:54,810 D 16281 16351] direct_actor_task_submitter.cc:423: Pushing task 4e1852778ed2b1e261393dc0e76fe0933af5942301000000 to actor 61393dc0e76fe0933af5942301000000 actor counter 209 seq no 209 num queued 0
[2022-07-20 22:48:54,810 D 16281 16363] reference_count.cc:316: Remove local reference 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000
[2022-07-20 22:48:54,810 D 16281 16363] reference_count.cc:317: REF 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000 borrowers: 0 local_ref_count: 0 submitted_count: 0 contained_in_owned: 0 contained_in_borrowed: 0 contains: 0 stored_in: 0 lineage_ref_count: 0
[2022-07-20 22:48:54,810 D 16281 16363] reference_count.cc:542: Attempting to delete object 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000
[2022-07-20 22:48:54,810 D 16281 16363] reference_count.cc:548: REF 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000 borrowers: 0 local_ref_count: 0 submitted_count: 0 contained_in_owned: 0 contained_in_borrowed: 0 contains: 0 stored_in: 0 lineage_ref_count: 0
[2022-07-20 22:48:54,810 D 16281 16363] reference_count.cc:585: Deleting Reference to object 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000
[2022-07-20 22:48:54,810 D 16281 16363] reference_count.cc:404: Releasing lineage for object 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000
[2022-07-20 22:48:54,810 D 16281 16363] task_manager.cc:547: No lineage for object 0a7603de59588c07fc54eb2aff11565b5138d2ad0100000002000000
[2022-07-20 22:48:54,824 D 16281 16351] task_manager.cc:226: Completing task 4e1852778ed2b1e261393dc0e76fe0933af5942301000000
....
[2022-07-20 22:48:54,974 D 16281 16363] reference_count.cc:200: Adding owned object 52314ba75263af075f40c38b4ea7145ed1c52be50100000002000000
[2022-07-20 22:48:54,974 D 16281 16363] reference_count.cc:335: Increment ref count for submitted task argument ffffffffffffffff5f40c38b4ea7145ed1c52be50100000001000000
[2022-07-20 22:48:54,974 D 16281 16363] direct_actor_task_submitter.cc:76: Submitting task 52314ba75263af075f40c38b4ea7145ed1c52be501000000
[2022-07-20 22:48:54,974 D 16281 16351] direct_actor_task_submitter.cc:423: Pushing task 52314ba75263af075f40c38b4ea7145ed1c52be501000000 to actor 5f40c38b4ea7145ed1c52be501000000 actor counter 210 seq no 210 num queued 0
[2022-07-20 22:48:54,975 D 16281 16351] direct_actor_task_submitter.cc:449: The task 288fc4e3db68adeafc54eb2aff11565b5138d2ad01000000 has already been marked as failed. Ingore the reply.
[2022-07-20 22:48:54,975 D 16281 16351] direct_actor_task_submitter.cc:449: The task 0a7603de59588c07fc54eb2aff11565b5138d2ad01000000 has already been marked as failed. Ingore the reply.
[2022-07-20 22:48:54,992 D 16281 16351] task_manager.cc:226: Completing task 52314ba75263af075f40c38b4ea7145ed1c52be501000000
[2022-07-20 22:48:54,992 D 16281 16351] reference_count.cc:1387: Published message for 52314ba75263af075f40c38b4ea7145ed1c52be50100000001000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: 3, and primary node ID: NIL_ID, pending creation? 1
...


```

**Comment by iycheng** [comment]
> The actor died notification is pushed late. The cause is about the GCS detect it late. Still checking about why.


GCS use heart beat to detect the failure. It's set to be 30s. So this behavior is expected right now.

**Comment by iycheng** [comment]
Another issue is that the client side is not aware of the server side shutdown. So the request is pending for a very long time.

**Comment by iycheng** [comment]
Finally figure out where does the 90s comes from. It comes from the serve and serve will kill it if it's not health

**Comment by iycheng** [comment]
Another possible mitigate is to improve the GCS <-> raylet heartbeat algorithm.

**Comment by iycheng** [comment]
```
I0722 03:05:20.343936615    9847 subchannel.cc:899]          subchannel 0x55c91d13cd00 {address=ipv4:172.31.58.40:45621, args={grpc.client_channel_factory=0x55c91c4ce940, grpc.default_authority=172.31.58.40:45621, grpc.enable_http_proxy=0, grpc.http2.max_pings_without_data=0, grpc.http2.min_ping_interval_without_data_ms=5000, grpc.http2.min_time_between_pings_ms=10000, grpc.internal.channel_credentials=0x55c91c4ce920, grpc.internal.security_connector=0x55c91cfcc830, grpc.internal.subchannel_pool=0x55c91c4d1e60, grpc.keepalive_permit_without_calls=1, grpc.keepalive_time_ms=10000, grpc.keepalive_timeout_ms=5000, grpc.max_receive_message_length=262144000, grpc.max_send_message_length=262144000, grpc.primary_user_agent=grpc-c++/1.48.0, grpc.resource_quota=0x55c91c47e740, grpc.server_uri=dns:///172.31.58.40:45621}}: backoff delay elapsed, reporting IDLE
```

This process actually is dead, but somehow grpc still thing it's activate.

**Comment by iycheng** [comment]
```
[2022-07-22 04:11:57,508 I 8934 9052] direct_actor_task_submitter.cc:444: DBG:::: GrpcUnavailable: RPC Error message: keepalive watchdog timeout; RPC Error details: . Cost: 1.28639e+09
[2022-07-22 04:11:57,508 I 8934 9052] task_manager.cc:432: Task failed: GrpcUnavailable: RPC Error message: keepalive watchdog timeout; RPC Error details: : Type=ACTOR_TASK, Language=PYTHON, Resources: {}, function_descriptor={type=PythonFunctionDescriptor, module_name=ray.serve.replica, class_name=ServeReplica:DAGDriver, function_name=handle_request, function_hash=}, task_id=827234262db55bd50f99744eead9f5eacbf81aab01000000, task_name=ServeReplica:DAGDriver.handle_request(), job_id=01000000, num_args=4, num_returns=3, depth=0, actor_task_spec={actor_id=0f99744eead9f5eacbf81aab01000000, actor_caller_id=ffffffffffffffff01c5831bf0730cdc7733463f01000000, actor_counter=25}
```

Set up everything and then this kind of error.


```
[2022-07-22 03:50:31,117 D 8934 9144] direct_actor_task_submitter.cc:76: Submitting task 827234262db55bd50f99744eead9f5eacbf81aab01000000
[2022-07-22 03:50:31,118 D 8934 9052] direct_actor_task_submitter.cc:423: Pushing task 827234262db55bd50f99744eead9f5eacbf81aab01000000 to actor 0f99744eead9f5eacbf81aab01000000 actor counter 25 seq no 25 num queued 1
[2022-07-22 04:11:57,508 D 8934 9052] task_manager.cc:459: Task attempt 827234262db55bd50f99744eead9f5eacbf81aab01000000 failed with error ACTOR_DIED
[2022-07-22 04:11:57,508 D 8934 9052] task_manager.cc:406: Task 827234262db55bd50f99744eead9f5eacbf81aab01000000 failed with error ACTOR_DIED
[2022-07-22 04:11:57,508 I 8934 9052] task_manager.cc:432: Task failed: GrpcUnavailable: RPC Error message: keepalive watchdog timeout; RPC Error details: : Type=ACTOR_TASK, Language=PYTHON, Resources: {}, function_descriptor={type=PythonFunctionDescriptor, module_name=ray.serve.replica, class_name=ServeReplica:DAGDriver, function_name=handle_request, function_hash=}, task_id=827234262db55bd50f99744eead9f5eacbf81aab01000000, task_name=ServeReplica:DAGDriver.handle_request(), job_id=01000000, num_args=4, num_returns=3, depth=0, actor_task_spec={actor_id=0f99744eead9f5eacbf81aab01000000, actor_caller_id=ffffffffffffffff01c5831bf0730cdc7733463f01000000, actor_counter=25}
[2022-07-22 04:11:57,508 D 8934 9052] reference_count.cc:1387: Published message for 827234262db55bd50f99744eead9f5eacbf81aab0100000001000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: -1, and primary node ID: NIL_ID, pending creation? 0
[2022-07-22 04:11:57,508 D 8934 9052] reference_count.cc:1387: Published message for 827234262db55bd50f99744eead9f5eacbf81aab0100000002000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and object size: -1, and primary node ID: NIL_ID, pending creation? 0

```

**Comment by iycheng** [comment]
```
[2022-07-22 03:49:19,798 I 10012 10403] client_call.h:316: DBG: STATUS: 0x55a0780bddb0  2       :
[2022-07-22 03:49:19,990 D 10012 10387] gcs_rpc_client.h:514: GCS channel status: 2
[2022-07-22 03:49:19,992 D 10012 10387] accessor.cc:784: Adding profile data, component type = worker, node id = 971b12252f31c4826d622b9078765442db595ffdc6ddf9a55e79bb04
[2022-07-22 03:49:19,992 D 10012 10387] profiling.cc:87: Pushed 4 events to GCS.
[2022-07-22 03:49:19,992 D 10012 10387] accessor.cc:795: Finished adding profile data, status = OK, component type = worker, node id = 971b12252f31c4826d622b9078765442db595ffdc6ddf9a55e79bb04
[2022-07-22 03:49:19,992 D 10012 10387] task_manager.cc:596: Treat task as failed. task_id: 18d2e2c1ab4488d4aa0778f5d4bb679317211ca201000000, error_type: ACTOR_DIED
[2022-07-22 03:49:19,992 D 10012 10387] memory_store.cc:201: Putting object into memory store. objectid is 18d2e2c1ab4488d4aa0778f5d4bb679317211ca20100000001000000
[2022-07-22 03:49:19,992 D 10012 10387] memory_store.cc:201: Putting object into memory store. objectid is 18d2e2c1ab4488d4aa0778f5d4bb679317211ca20100000002000000
[2022-07-22 03:49:19,992 D 10012 10387] reference_count.cc:444: Releasing ref for submitted task argument 18d2e2c1ab4488d4aa0778f5d4bb679317211ca20100000002000000
[2022-07-22 03:49:19,992 D 10012 10387] direct_actor_task_submitter.cc:423: Pushing task 157d2ce69a686dee16e47fd5a565150c7131872b01000000 to actor 16e47fd5a565150c7131872b01000000 actor counter 8 seq no 8
 num queued 0
[2022-07-22 03:49:19,993 D 10012 10387] memory_store.cc:201: Putting object into memory store. objectid is 18d2e2c1ab4488d4aa0778f5d4bb679317211ca20100000003000000
[2022-07-22 03:49:19,993 I 10012 10387] direct_actor_task_submitter.cc:444: DBG:::: GrpcUnavailable: RPC Error message: failed to connect to all addresses; last error: UNKNOWN: Failed to connect to remote
 host: Connection refused; RPC Error details: . Cost: 407.855
[2022-07-22 03:49:19,993 D 10012 10387] task_manager.cc:459: Task attempt 157d2ce69a686dee16e47fd5a565150c7131872b01000000 failed with error ACTOR_DIED
[2022-07-22 03:49:19,993 D 10012 10387] task_manager.cc:406: Task 157d2ce69a686dee16e47fd5a565150c7131872b01000000 failed with error ACTOR_DIED
[2022-07-22 03:49:19,993 I 10012 10387] task_manager.cc:432: Task failed: GrpcUnavailable: RPC Error message: failed to connect to all addresses; last error: UNKNOWN: Failed to connect to remote host: Con
nection refused; RPC Error details: : Type=ACTOR_TASK, Language=PYTHON, Resources: {}, function_descriptor={type=PythonFunctionDescriptor, module_name=ray.serve.replica, class_name=ServeReplica:e, functio
n_name=handle_request, function_hash=}, task_id=157d2ce69a686dee16e47fd5a565150c7131872b01000000, task_name=ServeReplica:e.handle_request(), job_id=01000000, num_args=4, num_returns=3, depth=0, actor_task
_spec={actor_id=16e47fd5a565150c7131872b01000000, actor_caller_id=ffffffffffffffff0f99744eead9f5eacbf81aab01000000, actor_counter=8}
[2022-07-22 03:49:19,993 D 10012 10387] reference_count.cc:1387: Published message for 157d2ce69a686dee16e47fd5a565150c7131872b0100000001000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and
object size: -1, and primary node ID: NIL_ID, pending creation? 0
[2022-07-22 03:49:19,993 D 10012 10387] reference_count.cc:1387: Published message for 157d2ce69a686dee16e47fd5a565150c7131872b0100000002000000, 0 locations, spilled url: [], spilled node ID: NIL_ID, and
object size: -1, and primary node ID: NIL_ID, pending creation? 0
[2022-07-22 03:49:19,993 D 10012 10387] reference_count.cc:864: Merging ref ffffffffffffffff16e47fd5a565150c7131872b0100000001000000
[2022-07-22 03:49:19,993 D 10012 10387] reference_count.cc:444: Releasing ref for submitted task argument ffffffffffffffff16e47fd5a565150c7131872b0100000001000000
[2022-07-22 03:49:19,993 I 10012 10387] direct_actor_task_submitter.cc:515: PushActorTask failed because of network error, this task will be stashed away and waiting for Death info from GCS, task_id=157d2
ce69a686dee16e47fd5a565150c7131872b01000000, wait queue size=1
[2022-07-22 03:49:20,049 I 10012 10403] client_call.h:314: --- DUMP --- :
[2022-07-22 03:49:20,049 I 10012 10403] client_call.h:316: DBG: STATUS: 0x55a0780bddb0  2       :
[2022-07-22 03:49:20,224 D 10012 10387] subscriber.cc:365: Long polling request has replied from 5c6f67b90ec19c350193311dcbddfb1ff27761af62f441867b9e1ad1
[2022-07-22 03:49:20,224 D 10012 10387] subscriber.cc:348: Make a long polling request to 5c6f67b90ec19c350193311dcbddfb1ff27761af62f441867b9e1ad1
[2022-07-22 03:49:20,224 I 10012 10387] actor_manager.cc:214: received notification on actor, state: DEAD, actor_id: 16e47fd5a565150c7131872b01000000, ip address: 172.31.58.40, port: 43315, worker_id: 3a7
2b78a0c7c7e9f4278dd991cf4c5ed63756f2e80c31822cad20c18, raylet_id: 14c7c4abeebc1b14c446288250555bafde90a8d3e2270b4822dd421a, num_restarts: 0, death context type=ActorDiedErrorContext
[2022-07-22 03:49:20,224 D 10012 10387] actor_manager.cc:328: Actor name cache is invalidated for the actor of name SERVE_REPLICA::e#lSAGjL namespace serve id 16e47fd5a565150c7131872b01000000
[2022-07-22 03:49:20,224 D 10012 10387] direct_actor_task_submitter.cc:237: Disconnecting from actor 16e47fd5a565150c7131872b01000000, death context type=ActorDiedErrorContext
[2022-07-22 03:49:20,224 I 10012 10387] direct_actor_task_submitter.cc:270: Failing pending tasks for actor 16e47fd5a565150c7131872b01000000 because the actor is already dead.
[2022-07-22 03:49:20,224 I 10012 10387] direct_actor_task_submitter.cc:291: Failing tasks waiting for death info, size=1, actor_id=16e47fd5a565150c7131872b01000000
[2022-07-22 03:49:20,224 D 10012 10387] task_manager.cc:596: Treat task as failed. task_id: 157d2ce69a686dee16e47fd5a565150c7131872b01000000, error_type: ACTOR_DIED
[2022-07-22 03:49:20,224 D 10012 10387] memory_store.cc:201: Putting object into memory store. objectid is 157d2ce69a686dee16e47fd5a565150c7131872b0100000001000000
[2022-07-22 03:49:20,224 D 10012 10387] memory_store.cc:201: Putting object into memory store. objectid is 157d2ce69a686dee16e47fd5a565150c7131872b0100000002000000
[2022-07-22 03:49:20,224 D 10012 10387] reference_count.cc:444: Releasing ref for submitted task argument 51f9858d94bccd1433db39b971548642bd414dd20100000002000000
[2022-07-22 03:49:20,224 D 10012 10387] reference_count.cc:444: Releasing ref for submitted task argument 6cb3357313cb171a8e0a3a88ec612e68429777930100000002000000
[2022-07-22 03:49:20,224 D 10012 10387] reference_count.cc:444: Releasing ref for submitted task argument 157d2ce69a686dee16e47fd5a565150c7131872b0100000002000000
[2022-07-22 03:49:20,300 I 10012 10403] client_call.h:314: --- DUMP --- :
```

Very wired the actor is alive, but CQ does nothing

**Comment by iycheng** [comment]
There are two issues here: 

1. why keepalive timeout after a long time (20m)
2. why the cq is looping forever without any useful info

**Comment by iycheng** [comment]
```
Thread 10 (Thread 0x7f1d5dffb700 (LWP 23525) "worker.io"):
#0  syscall () at ../sysdeps/unix/sysv/linux/x86_64/syscall.S:38
#1  0x00007f1d789a99de in absl::lts_20211102::synchronization_internal::Waiter::Wait(absl::lts_20211102::synchronization_internal::KernelTimeout) () from /mnt/yic/ray/python/ray/_raylet.so
#2  0x00007f1d789a9913 in AbslInternalPerThreadSemWait_lts_20211102 () from /mnt/yic/ray/python/ray/_raylet.so
#3  0x00007f1d789a8213 in absl::lts_20211102::Mutex::Block(absl::lts_20211102::base_internal::PerThreadSynch*) () from /mnt/yic/ray/python/ray/_raylet.so
#4  0x00007f1d789a8f94 in absl::lts_20211102::Mutex::LockSlowLoop(absl::lts_20211102::SynchWaitParams*, int) () from /mnt/yic/ray/python/ray/_raylet.so
#5  0x00007f1d789a91bc in absl::lts_20211102::Mutex::LockSlowWithDeadline(absl::lts_20211102::MuHowS const*, absl::lts_20211102::Condition const*, absl::lts_20211102::synchronization_internal::KernelTimeo
ut, int) () from /mnt/yic/ray/python/ray/_raylet.so
#6  0x00007f1d7801e2a0 in absl::lts_20211102::Mutex::LockSlow(absl::lts_20211102::MuHowS const*, absl::lts_20211102::Condition const*, int) () from /mnt/yic/ray/python/ray/_raylet.so
#7  0x00007f1d782562b2 in std::_Function_handler<void (ray::Status), ray::core::CoreWorkerDirectActorTaskSubmitter::SubmitTask(ray::TaskSpecification)::{lambda()#1}::operator()()::{lambda(ray::Status)#1}>::_M_invoke(std::_Any_data const&, ray::Status&&) () from /mnt/yic/ray/python/ray/_raylet.so
#8  0x00007f1d782593dd in std::_Function_handler<void (std::shared_ptr<ray::RayObject>), ray::core::LocalDependencyResolver::ResolveDependencies(ray::TaskSpecification&, std::function<void (ray::Status)>)::{lambda(std::shared_ptr<ray::RayObject>)#1}>::_M_invoke(std::_Any_data const&, std::shared_ptr<ray::RayObject>&&) () from /mnt/yic/ray/python/ray/_raylet.so
#9  0x00007f1d7827dd19 in ray::core::CoreWorkerMemoryStore::Put(ray::RayObject const&, ray::ObjectID const&) () from /mnt/yic/ray/python/ray/_raylet.so
#10 0x00007f1d7826ea22 in ray::core::TaskManager::MarkTaskReturnObjectsFailed(ray::TaskSpecification const&, ray::rpc::ErrorType, ray::rpc::RayErrorInfo const*) () from /mnt/yic/ray/python/ray/_raylet.so
#11 0x00007f1d78255fc4 in ray::core::CoreWorkerDirectActorTaskSubmitter::DisconnectActor(ray::ActorID const&, long, bool, ray::rpc::ActorDeathCause const&) () from /mnt/yic/ray/python/ray/_raylet.so
#12 0x00007f1d782a8336 in ray::core::ActorManager::HandleActorStateNotification(ray::ActorID const&, ray::rpc::ActorTableData const&) () from /mnt/yic/ray/python/ray/_raylet.so
#13 0x00007f1d7838e979 in std::_Function_handler<void (ray::rpc::PubMessage const&), ray::gcs::GcsSubscriber::SubscribeActor(ray::ActorID const&, std::function<void (ray::ActorID const&, ray::rpc::ActorTableData const&)> const&, std::function<void (ray::Status)> const&)::{lambda(ray::rpc::PubMessage const&)#1}>::_M_invoke(std::_Any_data const&, ray::rpc::PubMessage const&) () from /mnt/yic/ray/python/ray/_raylet.so
#14 0x00007f1d784ad139 in EventTracker::RecordExecution(std::function<void ()> const&, std::shared_ptr<StatsHandle>) () from /mnt/yic/ray/python/ray/_raylet.so
#15 0x00007f1d7846578b in std::_Function_handler<void (), instrumented_io_context::post(std::function<void ()>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)::{lambda()#1}>::_M_invoke(std::_Any_data const&) () from /mnt/yic/ray/python/ray/_raylet.so
#16 0x00007f1d78465aa8 in boost::asio::detail::completion_handler<std::function<void ()>, boost::asio::io_context::basic_executor_type<std::allocator<void>, 0u> >::do_complete(void*, boost::asio::detail::scheduler_operation*, boost::system::error_code const&, unsigned long) () from /mnt/yic/ray/python/ray/_raylet.so
#17 0x00007f1d78977ebc in boost::asio::detail::scheduler::do_run_one(boost::asio::detail::conditionally_enabled_mutex::scoped_lock&, boost::asio::detail::scheduler_thread_info&, boost::system::error_code const&) () from /mnt/yic/ray/python/ray/_raylet.so
#18 0x00007f1d78978d51 in boost::asio::detail::scheduler::run(boost::system::error_code&) () from /mnt/yic/ray/python/ray/_raylet.so
#19 0x00007f1d789790b0 in boost::asio::io_context::run() () from /mnt/yic/ray/python/ray/_raylet.so
#20 0x00007f1d781f16e5 in ray::core::CoreWorker::RunIOService() () from /mnt/yic/ray/python/ray/_raylet.so
#21 0x00007f1d778f2039 in std::execute_native_thread_routine (__p=0x55a709dce520) at /home/builder/ktietz/cos6/ci_cos6/ctng-compilers_1622658800915/work/.build/x86_64-conda-linux-gnu/src/gcc/libstdc++-v3/src/c++11/thread.cc:80
#22 0x00007f1d7976a6db in start_thread (arg=0x7f1d5dffb700) at pthread_create.c:463
#23 0x00007f1d7949361f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```
Log from the hanging actors. A deadlock seem to me.

**Comment by iycheng** [comment]
```
Thread 10 (Thread 0x7feb54ff9700 (LWP 8644) "worker.io"):
#0  __lll_lock_wait () at ../sysdeps/unix/sysv/linux/x86_64/lowlevellock.S:135
#1  0x00007feb687c5025 in __GI___pthread_mutex_lock (mutex=0x558be38333a0) at ../nptl/pthread_mutex_lock.c:80
#2  0x00007feb672ae31c in __gthread_mutex_lock (__mutex=0x558be38333a0) at /usr/include/x86_64-linux-gnu/c++/9/bits/gthr-default.h:749
#3  std::mutex::lock (this=0x558be38333a0) at /usr/include/c++/9/bits/std_mutex.h:100
#4  std::lock_guard<std::mutex>::lock_guard (__m=..., this=<synthetic pointer>) at /usr/include/c++/9/bits/std_mutex.h:159
#5  ray::core::CoreWorkerDirectActorTaskSubmitter::<lambda()>::<lambda(ray::Status)>::operator() (status=..., __closure=0x7feb300fc030) at src/ray/core_worker/transport/direct_actor_task_submitter.cc:107
#6  std::_Function_handler<void(ray::Status), ray::core::CoreWorkerDirectActorTaskSubmitter::SubmitTask(ray::TaskSpecification)::<lambda()> mutable::<lambda(ray::Status)> >::_M_invoke(const std::_Any_data &, ray::Status &&) (__functor=..., __args#0=...) at /usr/inclu
de/c++/9/bits/std_function.h:300
#7  0x00007feb672b14ad in std::function<void (ray::Status)>::operator()(ray::Status) const (__args#0=..., this=0x7fead0056078) at /usr/include/c++/9/bits/std_function.h:688
#8  ray::core::LocalDependencyResolver::<lambda(std::shared_ptr<ray::RayObject>)>::operator() (obj=..., __closure=<optimized out>) at src/ray/core_worker/transport/dependency_resolver.cc:133
#9  std::_Function_handler<void(std::shared_ptr<ray::RayObject>), ray::core::LocalDependencyResolver::ResolveDependencies(ray::TaskSpecification&, std::function<void(ray::Status)>)::<lambda(std::shared_ptr<ray::RayObject>)> >::_M_invoke(const std::_Any_data &, std::s
hared_ptr<ray::RayObject> &&) (__functor=..., __args#0=...) at /usr/include/c++/9/bits/std_function.h:300
#10 0x00007feb672d5de9 in std::function<void (std::shared_ptr<ray::RayObject>)>::operator()(std::shared_ptr<ray::RayObject>) const (__args#0=..., this=0x7feb300e8ce0) at /usr/include/c++/9/bits/std_function.h:688
#11 ray::core::CoreWorkerMemoryStore::Put (this=this@entry=0x558be2c7ea00, object=..., object_id=...) at src/ray/core_worker/store_provider/memory_store/memory_store.cc:261
#12 0x00007feb672c6af2 in ray::core::TaskManager::MarkTaskReturnObjectsFailed (this=0x558be38331d0, spec=..., error_type=ray::rpc::ACTOR_DIED, ray_error_info=0x7feb54ff8830) at src/ray/core_worker/task_manager.cc:604
#13 0x00007feb672adf83 in ray::core::CoreWorkerDirectActorTaskSubmitter::DisconnectActor (this=0x558be3833390, actor_id=..., num_restarts=<optimized out>, dead=<optimized out>, death_cause=...) at src/ray/core_worker/transport/direct_actor_task_submitter.cc:294
#14 0x00007feb67300406 in ray::core::ActorManager::HandleActorStateNotification (this=0x558be3833820, actor_id=..., actor_data=...) at bazel-out/k8-opt/bin/src/ray/protobuf/gcs.pb.h:7354
#15 0x00007feb673e6a49 in std::function<void (ray::ActorID const&, ray::rpc::ActorTableData const&)>::operator()(ray::ActorID const&, ray::rpc::ActorTableData const&) const (__args#1=..., __args#0=..., this=0x7feb30146018) at /usr/include/c++/9/bits/std_function.h:68
8
#16 ray::gcs::GcsSubscriber::<lambda(const ray::rpc::PubMessage&)>::operator() (msg=..., __closure=0x7feb30146000) at src/ray/gcs/pubsub/gcs_pub_sub.cc:129
#17 std::_Function_handler<void(const ray::rpc::PubMessage&), ray::gcs::GcsSubscriber::SubscribeActor(const ray::ActorID&, ray::gcs::SubscribeCallback<ray::ActorID, ray::rpc::ActorTableData>&, const StatusCallback&)::<lambda(const ray::rpc::PubMessage&)> >::_M_invoke
(const std::_Any_data &, const ray::rpc::PubMessage &) (__functor=..., __args#0=...) at /usr/include/c++/9/bits/std_function.h:300
#18 0x00007feb67505209 in std::function<void ()>::operator()() const (this=0x7feb30128860) at /usr/include/c++/9/bits/std_function.h:688
#19 EventTracker::RecordExecution(std::function<void ()> const&, std::shared_ptr<StatsHandle>) (fn=..., handle=...) at src/ray/common/event_stats.cc:90
#20 0x00007feb674bd85b in instrumented_io_context::<lambda()>::operator() (__closure=<optimized out>) at src/ray/common/asio/instrumented_io_context.cc:36
#21 std::_Function_handler<void(), instrumented_io_context::post(std::function<void()>, std::string)::<lambda()> >::_M_invoke(const std::_Any_data &) (__functor=...) at /usr/include/c++/9/bits/std_function.h:300
#22 0x00007feb674bdb78 in std::function<void ()>::operator()() const (this=0x7feb54ff8bd0) at /usr/include/c++/9/bits/std_function.h:688
#23 boost::asio::asio_handler_invoke<std::function<void ()> >(std::function<void ()>&, ...) (function=...) at external/boost/boost/asio/handler_invoke_hook.hpp:88
#24 boost_asio_handler_invoke_helpers::invoke<std::function<void ()>, std::function<void ()> >(std::function<void ()>&, std::function<void ()>&) (context=..., function=...) at external/boost/boost/asio/detail/handler_invoke_helpers.hpp:54
#25 boost::asio::detail::handler_work<std::function<void ()>, boost::asio::io_context::basic_executor_type<std::allocator<void>, 0u>, void>::complete<std::function<void ()> >(std::function<void ()>&, std::function<void ()>&) (this=<synthetic pointer>, handler=..., fu
nction=...) at external/boost/boost/asio/detail/handler_work.hpp:425
#26 boost::asio::detail::completion_handler<std::function<void ()>, boost::asio::io_context::basic_executor_type<std::allocator<void>, 0u> >::do_complete(void*, boost::asio::detail::scheduler_operation*, boost::system::error_code const&, unsigned long) (owner=0x558be
```

**Comment by iycheng** [comment]
The adhoc fix is working. I'll test in kuberay

**Comment by iycheng** [comment]
Closes with https://github.com/ray-project/ray/pull/26898
